plugins/catalog-backend/src/next/processing/DefaultCatalogProcessingOrchestrator.ts

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Entity,
+  EntityPolicy,
+  LocationEntity,
+  LocationSpec,
+  parseLocationReference,
+  stringifyEntityRef,
+} from '@backstage/catalog-model';
+import { ConflictError, InputError } from '@backstage/errors';
+import { ScmIntegrationRegistry } from '@backstage/integration';
+import path from 'path';
+import { Logger } from 'winston';
+import {
+  CatalogProcessor,
+  CatalogProcessorParser,
+} from '../../ingestion/processors';
+import * as results from '../../ingestion/processors/results';
+import {
+  CatalogProcessingOrchestrator,
+  EntityProcessingRequest,
+  EntityProcessingResult,
+} from './types';
+import { ProcessorOutputCollector } from './ProcessorOutputCollector';
+import {
+  getEntityLocationRef,
+  getEntityOriginLocationRef,
+  isLocationEntity,
+  toAbsoluteUrl,
+  validateEntity,
+  validateEntityEnvelope,
+} from './util';
+
+type Context = {
+  entityRef: string;
+  location: LocationSpec;
+  originLocation: LocationSpec;
+  collector: ProcessorOutputCollector;
+};
+
+export class DefaultCatalogProcessingOrchestrator
```

Comment:
This actually changes the structure a bit.

- Broke out a `ProcessorOutputCollector` instead of the `createEmitter` function at the bottom
- Moved out helper fns to `util.ts`
- Moved computed things into the `Context` type and the actual phases of processing into separate methods

The workflow is now a whole lot easier to follow IMO